test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -5711,5 +5711,76 @@ public void Dotnet_New_Template_Restore_Pack_Success(string template)
                 Assert.True(File.Exists(nupkgPath));
             }
         }
+
+
+        [Theory]
+        [InlineData("net472")]
+        [InlineData("net5.0")]
+        public void PackCommand_PackProject_PackageReference_PreReleaseDependency(string platform)
```

Comment:
you should add another test where there's two packages that generate warnings, and only one of them has the NoWarn. Pack should still warn/error on the second package.